lib/contextual_logger/context.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module ContextualLogger
+  module Context
+    def thread_context_for_logger_instance
+      # We include the object_id here to make these thread/fiber locals unique per logger instance.
+      @thread_context_for_logger_instance ||= "ContextualLogger::Context.context_for_#{object_id}".to_sym
```

Comment:
This is pretty sneaky. It's the only way I could think to get Thread/Fiber-locals to also be unique for each enclosing `Logger`. Notice that they won't get cleaned up when the `Logger` goes out of scope. I don't think it's likely that anyone would create a huge number of `Logger`s, so we should be fine.